.circleci/config.yml

```diff
@@ -93,4 +95,32 @@ jobs:
             if [ $? -ne 0 ]; then
               echo "Unformatted code detected. Please run black --fast . locally and commit to format your code." >&2
               exit 1
-            fi
\ No newline at end of file
+            fi
+
+  ## ------------------ Mypy for Static Type Checking python code ------------------
+  mypy-type-checks:
+    parameters:
+      with_merge: # Part of the PR-merge workflow
+        type: boolean
+        default: false
+    docker:
+      # https://circleci.com/developer/images/image/cimg/python
+      - image: cimg/python:3.11.0
+    steps:
+      - when:
+          condition: << parameters.with_merge >>
+          steps:
+            - disable_build_if_not_pr
+      - checkout
+      # Run Pyre type checking
+      - run:
+          name: Run Static type checking
+          command: |
+            set -x
+            pip wheel --wheel-dir=./wheel-dir -r requirements-type-checks.txt
+            pip install --find-links=./wheel-dir -r requirements-type-checks.txt
+            mypy $(git diff --name-only --diff-filter=A origin/master HEAD | grep '\.py$') && mypy
```

Comment:
What is being done in this step.  Add a comment explaining the diff and secondary && condition.

To make the information collected by `mypy` we should generate a report as well:  We can attach it as an artifact of the build.
```bash
mypy --html-report htmlmypy
Generated HTML report (via XSLT): /home/mklein0/Repositories/git/sightmachine/Code/sightmachine-sdk/htmlmypy/index.html
Success: no issues found in 3 source files
```
Though I am not sure how to interpret the report.   It says things are fine when marks them in red.

Reply:
I've added explanatory comments to the `config.yaml` file to clarify the steps involved. When using command line arguments with `mypy`, they will override the values specified in the configuration file. Consequently, we end up running `mypy` twice: once for the newly added files in the repository and then a second time for the files listed in `mypy.ini`.

The command `(git diff --name-only --diff-filter=A origin/master HEAD | grep '\.py$')` is simply provides a list of Python files that have been newly added between the `origin/master` branch and the current working commit.